src/sentry/utils/linksign.py

```diff
@@ -0,0 +1,62 @@
+from django.core import signing
+from django.core.urlresolvers import reverse
+
+from sentry import options
+from sentry.models import User
+from sentry.utils.http import absolute_uri
+from sentry.utils.numbers import base36_encode, base32_decode
+
+
+def get_signer():
+    return signing.TimestampSigner(salt='sentry-link-signature')
+
+
+def generate_signed_link(user, viewname, args=None, kwargs=None):
+    """This returns an absolute URL where the given user is signed in for
+    the given viewname with args and kwargs.  This returns a redirect link
+    that if followed sends the user to another URL which carries another
+    signature that is valid for that URL only.  The user can also be a user
+    ID.
+    """
+    if hasattr(user, 'is_authenticated'):
```

Comment:
Oh, it wasn't super clear then that this could also be a user id, and not a `User` model, but now reading the docstring, it _does_ say that. 👓 

Any reason to not check for `isinstance(user, (int, long))` instead? I feel like that'd be more clear on the intent. Or I'm overthinking.